rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,106 @@
+# Implementing CDEvents to Spinnaker using Java SDK
+
+
+| | |
+|-|-|
+| **Status**     | _**Proposed**, Accepted, Implemented, Obsolete_ |
+| **RFC #**      | https://github.com/spinnaker/governance/pull/299 |
+| **Author(s)**  | _Jalander Ramagiri (`@rjalander`)_|
+| **SIG / WG**   | CDF-CDEvents, CDF-SIGEvents |
+
+## Overview
+
+Many organisations use more than one CI/CD tools as part of their pipelines and most of the times the organisations implement their own glue code in order to make the tools work with each other.
+
+Implementing [CDEvents](https://cdevents.dev/) to Spinnaker enables Spinnaker to communicate with other CI/CD tools with in the ecosystem. [CDEvents](https://cdevents.dev/) is a common specification for Continuous Delivery events based on CloudEvents.
+Spinnaker can produce and consume [CDEvents](https://cdevents.dev/) using [Java SDK](https://github.com/cdevents/sdk-java) which is developed and maintained by the [CDEvents community](https://cdevents.dev/community/).
+
+### Goals and Non-Goals
+Goals:
+- Event-Driven Standardization with CI/CD tools like Keptn, Tekton, Jenkins, Spinnaker, etc..
```

Comment:
A question:  Has this integration been done in any of those tools?  Do we have a concrete implantation we could easily integrate with for integration tests/validation?

Reply:
We have a few integrations at POC level, and we are working in parallel to integrated CDEvents natively in other tools. In more details:

- Tekton natively supports sending and ingesting CloudEvents. We built an experimental version of that feature that uses CDEvents instead and we plan on migrating that to the latest version of CDEvents
- Jenkins also supports CloudEvents, but not CDEvents yet - there is a lot of interest in a CDEvents plugin for Jenkins though, Fidelity (a Jenkins user) is very keen on adopting CDEvents
- Keptn natively supports CloudEvents, even though part of their events are "imperative" which is not currently supported by CDEvents. We included Keptn in several POCs through external adapters between CDEvents and Keptn CloudEvents
- We are working with other tools communities as well, like Testkube, who are helping CDEvents improve test events
- We also had a POC with Flux and a POC with Argo thanks to @bradmccoydev